raft.c

```diff
@@ -1819,6 +1820,36 @@ static bool handleMultiExec(RedisRaftCtx *rr, RaftReq *req)
     return false;
 }
 
+void handleInfoCommand(RedisRaftCtx *rr, RaftReq *req) {
+    RaftRedisCommand *cmd = req->r.redis.cmds.commands[0];
+
+    const char * section = "all";
+
+    if (cmd->argc > 2) {
+        /* Note: we can't use RM_WrongArity here because our req->ctx is a thread-safe context
+         * with a synthetic client that no longer has the original argv.
+         */
+        RedisModule_ReplyWithError(req->ctx, "ERR wrong number of arguments for 'cluster' command");
+        goto exit;
+    }
+    if (cmd->argc == 2) {
+        section = RedisModule_StringPtrLen(cmd->argv[1], NULL);
+    }
+
+    RedisModuleCallReply *reply = RedisModule_Call(req->ctx, "INFO", "c", section);
+    size_t info_len;
+    const char *info = RedisModule_CallReplyProto(reply, &info_len);
+
+    char * pos = strstr(info, "cluster_enabled:0");
+    pos += 16;
+    *pos = '1';
+
+    RedisModule_ReplyWithStringBuffer(req->ctx, info, info_len);
```

Comment:
```suggestion
    char *pos = strstr(info, "cluster_enabled:");
    if (pos) {
        pos += strlen("cluster_enabled:");
        *pos = rr->config->sharding ? '1' : '0';
    }

    RedisModule_ReplyWithCallReply(req->ctx, reply);
    RedisModule_FreeCallReply(reply);
```



1 - If the command is something like "info cpu", reply string will not contain "cluster_enabled".  So, we need a null check here.
2 - Should we set `cluster_enabled` according to our `sharding` config? Don't know why do we need `sharding` config though. Maybe we can remove it and run as sharding enabled by default.

Reply:
I think we are leaning to have sharding always enabled.  my thought is that sharding yes will remain as an option but will enable the "gossipy" current mechanism for transmitting sharding information between clusters, and that will default to off once we have an external mechanism to set the sharding info via an external orchestration mechanism

Reply:
Ok, currently should we set `cluster_enabled` according to `sharding` config? If we set it `1` unconditionally and if `sharding` is no, that will cause a problem with the clients?  e.g We won't send responses for `cluster nodes` etc.